scripts/git-hooks/pre-commit

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Variable to check if we are in script mode of the pre-commit hook
+# This is used to determine if we should run the script in a specific mode
+# or if we are running it as a pre-commit hook
+# Get value from arguments or default to 0
+SCRIPT_MODE=${1:-0}
+
+if [ ${SCRIPT_MODE} -eq 1 ]; then
+  echo "Script mode is enabled. Running in pre-commit hook mode."
+fi
+
+FAILED=0
+
+# Colors for output
+if [ ${SCRIPT_MODE} -eq 1 ]; then
+  # Only set colors if output is a terminal
+  BOLD="\033[1m"
+  GREEN="\033[1;32m"
+  RED="\033[1;31m"
+  RESET="\033[0m"
+
+  GIT_CACHED=""
+else
+  # Fallback to no colors if not a terminal
+  BOLD=""
+  GREEN=""
+  RED=""
+  RESET=""
+
+  GIT_CACHED="--cached"
+fi
+
+echo "${BOLD}Running pre-commit hooks...${RESET}"
+
+# Check if there are staged changes in Python files
+PYTHON_STAGED_FILES=$(git diff ${GIT_CACHED} --name-only --diff-filter=ACM | grep -E "\.py$")
+if [ -n "$PYTHON_STAGED_FILES" ]; then
+  echo "${BOLD}Running backend linters...${RESET}"
+  make lint
+  BACKEND_RESULT=$?
+  if [ $BACKEND_RESULT -ne 0 ]; then
+    echo "${RED}Backend linting failed!${RESET}"
+    echo "${RED}Please fix the issues and try committing again.${RESET}"
```

Comment:
Looks like the RED isn't taken into account, even tho I don't see why not, because GREEN works
![image](https://github.com/user-attachments/assets/3846f5ee-6011-4f30-a6bb-cbe8a655742f)
****

Reply:
both !